src/Component/HomePage.js

```diff
@@ -1,14 +1,35 @@
-import React from 'react';
+import React from "react";
 import logo from "../images/Split-it-logo.png";
+import SignUpModal from "./SignUpModal";
+import SignInModal from "./SignInModal";
 
-function HomePage() {
-    return (
-        <div>
-            <img src={logo} alt="Spli-it app logo" />
-            <button>Log-in</button>
-            <button>Sign-up</button>
-        </div>
-    )
+class HomePage extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			isModalOpen: false
+        };
+        this.handleModalWindow = this.handleModalWindow.bind(this);
+    }
+      handleModalWindow = e => {
+        e.preventDefault();
+        this.setState(() => ({ isModalOpen: !this.state.isModalOpen }));
```

Comment:
For a better setState for this case follow [this guide ](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous).

Reply:
thanks for the doc. The only thing differently I did here is passing state as an argument in the callback function inside setState.